bluesky.py

```diff
@@ -0,0 +1,90 @@
+import models
+from granary import bluesky as gr_bluesky
+from oauth_dropins import bluesky as oauth_bluesky
+from flask_app import app
+import util
+import logging
+from flask import flash, render_template
+from oauth_dropins.webutil.util import json_loads
+from urllib.parse import quote
+
+logger = logging.getLogger(__name__)
+
+
+class Bluesky(models.Source):
+  """
+  A Bluesky account.
+  """
+  SHORT_NAME = 'bluesky'
+  GR_CLASS = gr_bluesky.Bluesky
+  OAUTH_START = oauth_bluesky.Start
+  AUTH_MODEL = oauth_bluesky.BlueskyAuth
+  URL_CANONICALIZER = util.UrlCanonicalizer(
+          domain=GR_CLASS.DOMAIN,
+          # Bluesky does not support HEAD requests.
+          redirects=False)
+
+  @staticmethod
+  def new(auth_entity, **kwargs):
+    """Creates and returns a :class:`Bluesky` entity.
+
+    Args:
+      auth_entity: :class:`oauth_bluesky.BlueskyAuth`
+      kwargs: property values
+    """
+    assert 'username' not in kwargs
+    assert 'id' not in kwargs
+    user = json_loads(auth_entity.user_json)
+    (_, app_password) = auth_entity.access_token()
+    gr_source = gr_bluesky.Bluesky(user.get('handle'), app_password=app_password)
+    actor = gr_source.user_to_actor(user)
+    return Bluesky(id=auth_entity.key_id(),
+                   username=auth_entity.key_id(),
```

Comment:
this should probably be handle, right?

Reply:
Are you saying the value for username should be the Bluesky handle rather than the DID, or I should be passing a kwarg `handle` into the constructor?

Reply:
I think I'm being stupid here and you definitely mean the former, will change

Reply:
yes! 😁